src/protagonist/Portal/Features/Batches/Requests/ParseCsv.cs

```diff
@@ -0,0 +1,198 @@
+    }
+    
+    public static ParseCsvResult Success()
+    {
+        return new ParseCsvResult { IsSuccess = true };
+    }
+}
+
+public class ParseCsvHandler : IRequestHandler<ParseCsv, ParseCsvResult>
+{
+    private readonly IDlcsClient dlcsClient;
+    private readonly ILogger<ParseCsvHandler> logger;
+    private readonly int customerId;
+    private readonly int maxBatchSize;
+    
+    public ParseCsvHandler(IDlcsClient dlcsClient, IOptions<PortalSettings> portalSettings, ILogger<ParseCsvHandler> logger, ClaimsPrincipal currentUser)
+    {
+        this.dlcsClient = dlcsClient;
+        this.logger = logger;
+        customerId = (currentUser.GetCustomerId() ?? -1);
+        maxBatchSize = portalSettings.Value.MaxBatchSize;
+    }
+
+    public async Task<ParseCsvResult> Handle(ParseCsv request, CancellationToken cancellationToken)
+    {
+        var distinctRows = new Dictionary<int, Image>();
+        var readErrors = new List<string>();
+        
+        using (var stream = new StreamReader(request.File.OpenReadStream()))
+        using (var csv = new CsvReader(stream, CultureInfo.InvariantCulture))
+        {
+            csv.Read();
+            csv.ReadHeader();
+            while (csv.Read())
+            {
+                try
+                {
+                    var record = csv.GetRecord<ImageIngestModel>();
+                    if (record.AssetType.ToLower() != "image") continue;
+                    if (distinctRows.ContainsKey(record.Line.Value)) continue;
+                    distinctRows.Add(record.Line.Value, new Image
+                    {
+                        CustomerId = customerId,
+                        ModelId = record.Id,
+                        Space = record.Space,
+                        String1 = record.String1,
+                        String2 = record.String2,
+                        String3 = record.String3,
+                        Origin = record.Origin,
+                        Tags = record.Tags.Split(",",
+                            StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries),
+                        Roles = record.Roles.Split(",",
+                            StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries),
+                        MaxUnauthorised = record.MaxUnauthorized ?? -1,
+                        Number1 = record.Number1,
+                        Number2 = record.Number2,
+                        Number3 = record.Number3,
+                        InitialOrigin = record.InitialOrigin,
+                        Family = AssetFamily.Image,
+                        MediaType = "image/jp2"
+                    });
+                }
+                catch (BadDataException badDataEx)
+                {
+                    return ParseCsvResult.Failure(
+                        $"CSV read error: bad data found in file at line {badDataEx.Context.Parser.Row}, row {badDataEx.Context.Reader.CurrentIndex}");
```

Comment:
These error messages all mention `line` but I think this is row index - rather than `Line` value specified in the spreadsheet. It might be worth using different wording to differentiate? e.g. `bad data found in file at row `

Reply:
Good idea, I'll replace this one with `bad data found in file at row X, column Y`